app/models/host_report.rb

```diff
@@ -55,4 +55,53 @@ def self.search_by_keyword(_key, operator, value)
       conditions: sanitize_sql_for_conditions(["host_reports.report_keyword_ids @> ?", "{#{keyword_ids.join(',')}}"]),
     }
   end
+
+  def self.report
+    @report = {} if @report.nil?
+    @report.update(
+      :total_hosts => Host.all.count,
+      :change_hosts => report_changes.count,
+      :nochange_hosts => report_nochanges.count,
+      :failure_hosts => report_failures.count,
+      :disabled_hosts => disabled_alerts.count,
+      :reports_missing => report_missing.count
+    )
+  end
+
+  def self.report_missing
+    Host.search_for('not has last_report and status.enabled = true')
+  end
+
+  def self.disabled_alerts
+    hosts.where(:enabled => true)
+  end
+
+  def self.report_changes
+    hosts.search_for("report_changes = 1")
```

Comment:
If you can avoid scoped search, avoid it. It creates ugly SQL queries. I think in this case this could be a simple join.

Also report_changes is an alias to host_report.change which is an integer. So you are searching for `change > 0` not `= 1`. The same below.

Reply:
Still TODO.